src/sys/statfs.rs

```diff
@@ -88,31 +88,41 @@ impl Statfs {
         FsType(self.0.f_type)
     }
 
-
     /// Optimal transfer block size
     #[cfg(any(target_os = "ios", target_os = "macos"))]
     pub fn optimal_transfer_size(&self) -> libc::int32_t {
         self.0.f_iosize
     }
 
     /// Optimal transfer block size
-    #[cfg(target_os = "android")]
+    #[cfg(target_arch = "s390x")]
+    pub fn optimal_transfer_size(&self) -> libc::uint32_t {
+        self.0.f_bsize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(any(target_os = "android", all(target_os = "linux", target_env = "musl")))]
     pub fn optimal_transfer_size(&self) -> libc::c_ulong {
         self.0.f_bsize
     }
 
     /// Optimal transfer block size
-    #[cfg(target_os = "linux")]
+    #[cfg(all(target_os = "linux", not(target_env = "musl"), not(target_arch = "s390x")))]
     pub fn optimal_transfer_size(&self) -> libc::c_long {
         self.0.f_bsize
     }
 
     /// Optimal transfer block size
-    #[cfg(not(any(target_os = "linux", target_os = "android", target_os = "ios", target_os = "macos")))]
-    pub fn optimal_transfer_size(&self) -> libc::c_ulong {
+    #[cfg(target_os = "freebsd")]
+    pub fn optimal_transfer_size(&self) -> libc::uint64_t {
         self.0.f_iosize
     }
 
+    /// Optimal transfer block size
+    #[cfg(not(any(target_os = "linux", target_os = "android", target_os = "ios", target_os = "macos", target_os = "freebsd")))]
```

Comment:
This guard would be clearer expressed in the positive sense than in the negative.